tests/python/pants_test/rules/test_filedeps.py

```diff
@@ -0,0 +1,282 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import os
+
+from future.utils import text_type
+from pex.orderedset import OrderedSet
+
+from pants.build_graph.address import Address, BuildFileAddress
+from pants.engine.addressable import BuildFileAddresses
+from pants.engine.fs import PathGlobsAndRoot, PathGlobs
+from pants.engine.legacy.graph import HydratedTarget, TransitiveHydratedTargets
+from pants.engine.legacy.structs import TargetAdaptor
+from pants.rules.core.filedeps import file_deps
+from pants.util.contextutil import temporary_dir
+from pants.util.dirutil import safe_file_dump
+from pants.util.meta import AbstractClass
+from pants.util.objects import datatype
+from pants_test.engine.scheduler_test_base import SchedulerTestBase
+from pants_test.engine.util import MockConsole, run_rule
+from pants_test.test_base import TestBase
+
+
+class FileDepsTest(TestBase, SchedulerTestBase, AbstractClass):
+
+  def filedeps_rule_test(self, target_addresses, transitive_targets, expected_console_output):
+    console = MockConsole()
+
+    run_rule(file_deps, console, BuildFileAddresses(target_addresses), {
+      (TransitiveHydratedTargets, BuildFileAddresses): lambda _: transitive_targets,
+    })
+
+    self.assertEquals(console.stdout.getvalue(), expected_console_output)
+
+  def make_build_target_address(self, spec):
+    address = Address.parse(spec)
+    
+    return BuildFileAddress(
+      build_file=None,
+      target_name=address.target_name,
+      rel_path='{}/BUILD'.format(address.spec_path),
+    )
+
+  def make_build_target_address_with_ext(self, spec):
```

Comment:
You could maybe merge this method with `make_build_target_address` by giving `make_build_target_address` a default `ext` parameter, and then including it in your format string:

```
def make_build_target_address(self, spec, ext=''):
  ..
  '{}/BUILD{}'.format(address.spec_path, ext)
  ..
```